flow.py

```diff
@@ -57,17 +68,85 @@ def dest(self):
     def dest(self, dest):
         raise AttributError("Cannot modify a flow's destination node.")
 
-    def make_packet(self, id, src, dest, data):
+    def check_last(self):
+        """Method that checks if all packets have been acknowledged and
+        updating the NetworkSimulator object accordingly
+        """
+        if len(self.unacknowledged) == 0:
+            self.ns.decrement_active_flows()
+
+    def update_flow(a_packet):
+        """Upon receiving an acknowledgement packet, updating the flow's
+        attributes
+
+        Args:
+            a_packet (AcknowledgementPacket): Packet being sent
+                back from host
+        """
+        print("Flow: updated with acknowledgement packet: ", a_packet.id,
+            "And flow", self.flow_id)
+        self.check_last()
+        self.last_acknowledged = a_packet.packet_id
+        self.unacknowledged_packets.remove(a_packet.packet_id)
+
+    def send_packet(self):
+        """Method sends packets by first going through unacknowledged_packets
+
+        """
```

Comment:
if this can fit on the same line, i think it is supposed to. also the style guide says we should end phrases with periods.